source/common/upstream/upstream_impl.h

```diff
@@ -75,7 +77,8 @@ class HostDescriptionImpl : virtual public HostDescription {
                     .bool_value()),
         metadata_(std::make_shared<envoy::api::v2::core::Metadata>(metadata)), locality_(locality),
         locality_zone_stat_name_(locality.zone(), cluster->statsScope().symbolTable()),
-        priority_(priority) {
+        priority_(priority), socket_factory_(resolveTransportSocketFactory(
+                                 dest_address ? dest_address->asString() : "empty", metadata)) {
```

Comment:
I will move the logic ` dest_address ? dest_address->asString() : "empty"` into `resolveTransportSocketFactory`, so it is not evaluated other than debug logging is enabled.

Reply:
done. interesting to see the power of the ENVOY_LOG macro.